### PR TITLE
[baseimage] set default locale en_US.UTF-8

### PR DESCRIPTION
**Pull request by richard28530**
Signed-off-by: chenhu <chenhu@didichuxing.com>

**- What I did**
set default locale en_US.UTF-8
to avoid client (SSH) inject locale while login to switch 

**- How I did it**

**- How to verify it**
login from a machine with non en_US.UTF-8
verify 'env' command output to check 'LANG' environment variable
use 'TAB' to do the auto-completion
  
**- Description for the changelog**

### CODE REVIEW

**Comment by qiluo-msft** [comment]
We are conservative on disk space, so we originally remove the locale files. I found another solution to solve the issue https://github.com/Azure/sonic-buildimage/issues/1681 without any new packages.

You just need to create a new configuration file like:
```
sudo vi /etc/profile.d/locale.sh
export LANG=C
export LANGUAGE=C
```
logout and login. You will have no LANG injection from ssh client.
However, I notice the Debian Stretch bash prompt's color is lost. I don't find the clue right now...

**Comment by lguohan** [comment]
how large is the locale file?

**Comment by richard28530** [comment]
/usr/share/locale$ du -d 0 -h
50M	.

**Comment by sergeypopovich-ord** [comment]
How about localepurge Debian package? It's purpose to remove all unused locale message files and other stuff. While it is a hack it might be very useful to conserve space:

  # localepurge
  localepurge: Disk space freed in /usr/share/locale: 86292 KiB
  localepurge: Disk space freed in /usr/share/man: 7088 KiB

  Total disk space freed by localepurge: 93380 KiB

On test system after installation and manual run.

Also I think it is a good idea to run update-locale LANG=en_US.UTF-8 after we locale-gen en_US.UTF-8 to update /etc/default/locale.

**Comment by richard28530** [comment]
/usr/share/locale

File: build_debian.sh [hunk 1]
```diff
@@ -227,7 +227,14 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
     hping3                  \
     python-scapy            \
     tcptraceroute           \
-    mtr-tiny
+    mtr-tiny                \
+    locales
+
+#Adds a locale to a debian system in non-interactive mode
+sudo sed -i '/^#.* en_US.* /s/^#//' $FILESYSTEM_ROOT/etc/locale.gen && \
+    sudo sed -i "\$aLANG=en_US.UTF-8" $FILESYSTEM_ROOT/etc/default/locale && \
```
Comment (lguohan) [inline build_debian.sh]: maybe it is better to use update-locale after locale-gen here?

https://www.thomas-krenn.com/en/wiki/Configure_Locales_in_Ubuntu

**Comment by lguohan** [comment]
looks like the disk is not an issue, what about the update-locale suggested by sergey?

**Comment by richard28530** [comment]
add update-locale after manually modify it

I tried this below, doesn't work
<pre>
#Adds a locale to a debian system in non-interactive mode
sudo sed -i '/^#.* en_US.* /s/^#//' $FILESYSTEM_ROOT/etc/locale.gen && \
    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT locale-gen "en_US.UTF-8"
sudo LANG=en_US.UTF-8 DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT update-locale "en_US.UTF-8"
sudo LANG=C chroot $FILESYSTEM_ROOT bash -c "find /usr/share/i18n/locales/ ! -name 'en_US' -type f -exec rm -f {} +"
</pre>

**Comment by lguohan** [comment]
what is the error message?

**Comment by lguohan** [comment]
update-locale "LANG=en_US.UTF-8" is the correct command.

**Pull request merged**